### ISSUE TITLE
Override function focusInvalid

### ISSUE THREAD

**Issue by subsecond** [jzaefferer/jquery-validation#1456]
As discussed on [StackOverflow](http://stackoverflow.com/questions/29589717/override-function-focusinvalid-in-jquery-validation-plugin) I am trying to override the default behaviour of *focusInvalid*. When I try to use my own *invalidHandler* to focus on the first invalid element in a form, all of a sudden the form can be submitted by pressing the ENTER key.

Steps to reproduce:
* Launch https://jsfiddle.net/kfopux67/3/
* Enter a valid email address
* Press the ENTER key

Basically I am trying to go from the default:

```javascript
focusInvalid: function() {
	if ( this.settings.focusInvalid ) {
		try {
			$( this.findLastActive() || this.errorList.length && this.errorList[ 0 ].element || [])
			.filter( ":visible" )
			.focus()
			// manually trigger focusin event; without it, focusin handler isn't called, findLastActive won't have anything to find
			.trigger( "focusin" );
		} catch ( e ) {
			// ignore IE throwing errors when focusing hidden elements
		}
	}
},
```
...to something like...

```javascript
focusInvalid: function() {
    if ( this.settings.focusInvalid ) {
        try {
            var firstInvalidElement = $(this.errorList[0].element);
            $('html,body').scrollTop(firstInvalidElement.offset().top);
            firstInvalidElement.focus();
        } catch ( e ) {
            // ignore IE throwing errors when focusing hidden elements
        }
    }
}
```

...by changing the *invalidHandler* to something like:

```javascript
invalidHandler: function(form, validator) {
    var errors = validator.numberOfInvalids();
    if (errors) {                    
        var firstInvalidElement = $(validator.errorList[0].element);
        $('html,body').scrollTop(firstInvalidElement.offset().top);
        firstInvalidElement.focus();
    }
},
```

Is there - as assumed in the StackOverflow discussion - an issue in the code that allows submitting the form using the ENTER key or did I do something wrong in my adaptations?

**Comment by jzaefferer** [comment]
I'm sorry for the lack of activity on this issue. Instead of leaving it open any longer, I decided to close old issues without trying to address them, to longer give the false impression that it will get addressed eventually, especially after several years with no activity. It doesn't mean I'm abandoning the project, just that I'm unable to work through 200+ open issues with the little time I can afford to spend on this project.

To the reporter (or anyone else interested in this issue): If you're still affected by the same issue, please consider opening a new issue, with a testpage that demonstrates the issue with a current version of the plugin. Even better, make an attempt to fix the issue yourself, and improve the project by sending a pull request. This may seem daunting at first, but you'll likely learn some useful skills that you can apply elsewhere as well. And you can help keep this project alive. [We've documented how to do these things, too](https://github.com/jzaefferer/jquery-validation/blob/master/CONTRIBUTING.md). A patch is worth a thousand issues!

**Comment by bytestream** [comment]
@staabm any insight on this based on what I referenced above?

For arguments sake, I've updated @subsecond jsfiddle to the latest release (1.17.0)

Using `.focus()` in the `invalidHandler`: https://jsfiddle.net/kfopux67/51/
Using `focusInvalid: true`: https://jsfiddle.net/kfopux67/52/

To replicate:
1. Type `test@test.com` into email field
2. Press ENTER key

Actual Result:
Form submits by itself

Expected Result:
Validation message shows stating password is required
